Sampling based planning

avalenzu opened the pull request:
As demonstrated in lab meeting on Feb. 12. This pull request adds RRT and RRT-connect for $R^n$, SO(3), SE(3), RBM configuration space, and RBM task space (with kinematic constraints).

Adds tests 3D piano mover tests and Atlas board extraction test.

RobotLocomotionBuild commented:
Site: drake006
Build Name: master-pull-avalenzu-sampling-based-planning-into-RobotLocomotion-master
Build Server Directory: /tmp/drake-master-7375/drake
Pull Request: https://github.com/RobotLocomotion/drake/pull/760
CDash: http://kobol.csail.mit.edu/cd/viewTest.php?onlyfailed&buildid=10952

Most recent commits to drake included in this build:
```
962a727 Merge branch 'sampling-based-planning' of https://github.com/avalenzu/drake
74a1c29 Merge pull request #756 from RussTedrake/playing_w_cars
ab6c084 removed extra lines (as suggest by @avalenzu)
8983e28 Adds testTaskSpaceRRT.
fb84f4c Add Task and Configuration space tree classes.
```

359 tests run:
 346 passed
 0 cancelled (with missing dependencies )
 0 timed out
 13 failed
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/18\">known issue #18</a>: Singularities in the euler-angle floating base joint cause infrequent errors on unit tests with random initial conditions
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/286\">known issue #286</a>: getPoints from stl mesh file can fail in parallel unit tests
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/310\">known issue #310</a>: testLCMPlant fails on parallel build because it can't keep up with realtime
  4 <a href=\"https://github.com/RobotLocomotion/drake/issues/374\">known issue #374</a>: failing imu unit tests
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/466\">known issue #466</a>: implement Jdot*v for drakeFunction.kinematic methods
  5 unknown

solvers/test/rrt_basic (unknown issue)
```
error_id = MATLAB:UndefinedFunction
error_message = Undefined function 'VertexArrayTree' for input arguments of type 'double'.
is_segfault = 0
```
```Matlab

                            < M A T L A B (R) >
                  Copyright 1984-2012 The MathWorks, Inc.
                    R2012b (8.0.0.783) 64-bit (glnxa64)
                              August 22, 2012

 
To get started, type one of these: helpwin, helpdesk, or demo.
For product information, visit www.mathworks.com.
 
To reproduce this test use rng(414494150,'twister')
 
[Warning: Disabling autosave for simulink blocks (to avoid generating a lot of
*.mdl.autosave files in your directory.  If you aren't a regular Simulink user
and don't want this disabled, comment out this section in addpath_drake.
To disable this warning in the future, add
   warning('off','Drake:DisablingSimulinkAutosave')
to your matlab startup.m] 
[> In addpath_drake at 74
  In addpath_drake at 4] 
Undefined function 'VertexArrayTree' for input arguments of type 'double'.

Error in rrt_basic (line 8)
T = VertexArrayTree(2);

Error in fevalPackageSafe (line 21)
  feval(f);

 
<test_name>solvers/test/rrt_basic</test_name> <error_id>MATLAB:UndefinedFunction</error_id> <error_message>Undefined function 'VertexArrayTree' for input arguments of type 'double'.</error_message> 


```

solvers/test/rrt_bugtrap (unknown issue)
```
error_id = MATLAB:nonStrucReference
error_message = Attempt to reference field of non-structure array.
is_segfault = 0
```
```Matlab

                            < M A T L A B (R) >
                  Copyright 1984-2012 The MathWorks, Inc.
                    R2012b (8.0.0.783) 64-bit (glnxa64)
                              August 22, 2012

 
To get started, type one of these: helpwin, helpdesk, or demo.
For product information, visit www.mathworks.com.
 
To reproduce this test use rng(414494376,'twister')
 
[Warning: Disabling autosave for simulink blocks (to avoid generating a lot of
*.mdl.autosave files in your directory.  If you aren't a regular Simulink user
and don't want this disabled, comment out this section in addpath_drake.
To disable this warning in the future, add
   warning('off','Drake:DisablingSimulinkAutosave')
to your matlab startup.m] 
[> In addpath_drake at 74
  In addpath_drake at 4] 
Attempt to reference field of non-structure array.

Error in MotionPlanningTree/addPath (line 262)
        [obj, id_parent] = obj.addVertex(T_other.getVertex(id), id_parent);

Error in rrt_bugtrap (line 28)
      [TA, id_last] = TA.addPath(TB.extractPath(fliplr(path_ids_B(1:end-1))), path_ids_A(end));

Error in fevalPackageSafe (line 21)
  feval(f);

 
<test_name>solvers/test/rrt_bugtrap</test_name> <error_id>MATLAB:nonStrucReference</error_id> <error_message>Attempt to reference field of

**This post was going to be too long and has been truncated**

RobotLocomotionBuild commented:
Site: DRAKE003
Build Name: precompiled-pull-avalenzu-sampling-based-planning-into-RobotLocomotion-master
Build Server Directory: C:\\cygwin64\	mp\\drake-precompiled-26162\\drake
Pull Request: https://github.com/RobotLocomotion/drake/pull/760
CDash: http://kobol.csail.mit.edu/cd/viewTest.php?onlyfailed&buildid=10955

Most recent commits to drake included in this build:
```
033bed9 Merge branch 'sampling-based-planning' of https://github.com/avalenzu/drake
74a1c29 Merge pull request #756 from RussTedrake/playing_w_cars
ab6c084 removed extra lines (as suggest by @avalenzu)
8983e28 Adds testTaskSpaceRRT.
fb84f4c Add Task and Configuration space tree classes.
```

352 tests run:
 249 passed
 67 cancelled (with missing dependencies  SDPSolver:9, avl:4, bullet:9, gloptipoly3:3, gurobi:8, iris:3, lcm:23, mosek:2, octomap:1, sedumi:3, yalmip:2)
 0 timed out
 36 failed
  4 <a href=\"https://github.com/RobotLocomotion/drake/issues/374\">known issue #374</a>: failing imu unit tests
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/466\">known issue #466</a>: implement Jdot*v for drakeFunction.kinematic methods
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/468\">known issue #468</a>: non-bullet builds can throw logical error in active_collision parsing in collisionDetect
  25 <a href=\"https://github.com/RobotLocomotion/drake/issues/569\">known issue #569</a>: lcppath.mexw64 and lcppath.mexw32 fails to load on win machines
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/585\">known issue #585</a>: testIndividualCentersOfPressure fails on win64
  4 unknown

solvers/test/rrt_basic (unknown issue)
```
error_id = MATLAB:UndefinedFunction
error_message = Undefined function 'VertexArrayTree' for input arguments of type 'double'.
is_segfault = 0
```
```Matlab
C:/Program Files/MATLAB/R2012a/bin/win64/MATLAB.exe
 
To get started, type one of these: helpwin, helpdesk, or demo.
For product information, visit www.mathworks.com.
 
To reproduce this test use rng(414940204,'twister')
 
{Warning: Disabling autosave for simulink blocks (to avoid generating a lot of *.mdl.autosave files in your
directory.  If you aren't a regular Simulink user and don't want this disabled, comment out this section
in addpath_drake.
To disable this warning in the future, add
   warning('off','Drake:DisablingSimulinkAutosave')
to your matlab startup.m} 
{> In addpath_drake at 74
  In addpath_drake at 4} 
Undefined function 'VertexArrayTree' for input arguments of type 'double'.

Error in rrt_basic (line 8)
T = VertexArrayTree(2);

Error in fevalPackageSafe (line 21)
  feval(f);

 
<test_name>solvers/test/rrt_basic</test_name> <error_id>MATLAB:UndefinedFunction</error_id> <error_message>Undefined function 'VertexArrayTree' for input arguments of type 'double'.</error_message> 


```

solvers/test/rrt_bugtrap (unknown issue)
```
error_id = MATLAB:nonStrucReference
error_message = Attempt to reference field of non-structure array.
is_segfault = 0
```
```Matlab
C:/Program Files/MATLAB/R2012a/bin/win64/MATLAB.exe
 
To get started, type one of these: helpwin, helpdesk, or demo.
For product information, visit www.mathworks.com.
 
To reproduce this test use rng(414941465,'twister')
 
{Warning: Disabling autosave for simulink blocks (to avoid generating a lot of *.mdl.autosave files in your
directory.  If you aren't a regular Simulink user and don't want this disabled, comment out this section
in addpath_drake.
To disable this warning in the future, add
   warning('off','Drake:DisablingSimulinkAutosave')
to your matlab startup.m} 
{> In addpath_drake at 74
  In addpath_drake at 4} 
Attempt to reference field of non-structure array.

Error in MotionPlanningTree/addPath (line 262)
        [obj, id_parent] = obj.addVertex(T_other.getVertex(id), id_parent);

Error in rrt_bugtrap (line 28)
      [TA, id_last] = TA.addPath(TB.extractPath(fliplr(path_ids_B(1:end-1))), path_ids_A(end));

Error in fevalPackageSafe (line 21)
  feval(f);

 
<test_name>solvers/test/rrt_bugtrap</test_name> <error_id>MATLAB:nonStrucReference</error_id> <error_message>Attempt to reference field of non-structure array.</error_message> 


```

solvers/test/rrt_piano_mover_rotation (unknown issue)
```
error_id = MATLAB:nonLogicalConditional
error_message = Operands to the || and && operators must be convertible to logical sca

**This post was going to be too long and has been truncated**

avalenzu commented:
Sorry, I clearly missed some things in final preparations for the pull request. Will reopen when my tests pass.

avalenzu commented:
My new tests pass now. Re-opening.

RobotLocomotionBuild commented:
Site: drake006
Build Name: master-pull-avalenzu-sampling-based-planning-into-RobotLocomotion-master
Build Server Directory: /tmp/drake-master-32538/drake
Pull Request: https://github.com/RobotLocomotion/drake/pull/760
CDash: http://kobol.csail.mit.edu/cd/viewTest.php?onlyfailed&buildid=10958

Most recent commits to drake included in this build:
```
ddab137 Merge branch 'sampling-based-planning' of https://github.com/avalenzu/drake
fdca11f Fix tests
74a1c29 Merge pull request #756 from RussTedrake/playing_w_cars
ab6c084 removed extra lines (as suggest by @avalenzu)
8983e28 Adds testTaskSpaceRRT.
```

359 tests run:
 348 passed
 0 cancelled (with missing dependencies )
 0 timed out
 11 failed
  2 <a href=\"https://github.com/RobotLocomotion/drake/issues/18\">known issue #18</a>: Singularities in the euler-angle floating base joint cause infrequent errors on unit tests with random initial conditions
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/311\">known issue #311</a>: testFloatingBase test can fail to match tolerance
  4 <a href=\"https://github.com/RobotLocomotion/drake/issues/374\">known issue #374</a>: failing imu unit tests
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/466\">known issue #466</a>: implement Jdot*v for drakeFunction.kinematic methods
  3 unknown

solvers/test/rrt_piano_mover (unknown issue)
```
error_id = MATLAB:UndefinedFunction
error_message = Undefined function 'quaternionDistance' for input arguments of type 'double'.
is_segfault = 0
```
```Matlab

                            < M A T L A B (R) >
                  Copyright 1984-2012 The MathWorks, Inc.
                    R2012b (8.0.0.783) 64-bit (glnxa64)
                              August 22, 2012

 
To get started, type one of these: helpwin, helpdesk, or demo.
For product information, visit www.mathworks.com.
 
To reproduce this test use rng(415524980,'twister')
 
[Warning: Disabling autosave for simulink blocks (to avoid generating a lot of
*.mdl.autosave files in your directory.  If you aren't a regular Simulink user
and don't want this disabled, comment out this section in addpath_drake.
To disable this warning in the future, add
   warning('off','Drake:DisablingSimulinkAutosave')
to your matlab startup.m] 
[> In addpath_drake at 74
  In addpath_drake at 4] 
 Added the lcm jar to your javaclasspath (found via cmake)
 Added the lcmgl jar to your javaclasspath (found via cmake)
LCM: Disabling IPV6 support
LCM: TTL set to zero, traffic will not leave localhost.
Undefined function 'quaternionDistance' for input arguments of type 'double'.

Error in SO3MotionPlanningTree/distanceMetric (line 18)
      d = quaternionDistance(quat1, quat2);

Error in CompositeVertexArrayTree/distanceMetric (line 45)
          d = d + obj.weights(i)*obj.trees{i}.distanceMetric(q1(obj.idx{i},:), ...

Error in MotionPlanningTree/nearestNeighbor (line 243)
      d_all = obj.distanceMetric(q, obj.getVertex(1:obj.n));

Error in MotionPlanningTree/newConfig (line 131)
      [d, id_near] = nearestNeighbor(obj, q);

Error in MotionPlanningTree/extend (line 161)
      [obj, q_new, id_near] = newConfig(obj, q);

Error in MotionPlanningTree/rrtNew (line 48)
        [obj, status] = extend(obj, x_sample);

Error in rrt_piano_mover (line 57)
      [TA, path_ids_A, info] = TA.rrtNew(q0, qf, options);

Error in fevalPackageSafe (line 21)
  feval(f);

 
<test_name>solvers/test/rrt_piano_mover</test_name> <error_id>MATLAB:UndefinedFunction</error_id> <error_message>Undefined function 'quaternionDistance' for input arguments of type 'double'.</error_message> 


```

solvers/test/rrt_piano_mover_rotation (unknown issue)
```
error_id = MATLAB:UndefinedFunction
error_message = Undefined function 'quaternionDistance' for input arguments of type 'double'.
is_segfault = 0
```
```Matlab

                            < M A T L A B (R) >
                  Copyright 1984-2012 The MathWorks, Inc.
                    R2012b (8.0.0.783) 64-bit (glnxa64)
                              August 22, 2012

 
To get started, type one of these: helpwin, helpdesk, or demo.
For product information, visit www.mathworks.com.
 
To reproduce this test use rng(415525064,'twister')
 
[Warning: Disabling autosave for simulink blocks (to avoid generating a lot of
*.mdl.autosave files in your directory.  If you aren't a regular Simulink user
a

**This post was going to be too long and has been truncated**

RobotLocomotionBuild commented:
Site: DRAKE003
Build Name: precompiled-pull-avalenzu-sampling-based-planning-into-RobotLocomotion-master
Build Server Directory: C:\\cygwin64\	mp\\drake-precompiled-20657\\drake
Pull Request: https://github.com/RobotLocomotion/drake/pull/760
CDash: http://kobol.csail.mit.edu/cd/viewTest.php?onlyfailed&buildid=10961

Most recent commits to drake included in this build:
```
5f91127 Merge branch 'sampling-based-planning' of https://github.com/avalenzu/drake
aa52b79 Merge pull request #745 from psiorx/jc-profiler-tweaks
fdca11f Fix tests
74a1c29 Merge pull request #756 from RussTedrake/playing_w_cars
ab6c084 removed extra lines (as suggest by @avalenzu)
```

352 tests run:
 251 passed
 68 cancelled (with missing dependencies  SDPSolver:9, avl:4, bullet:9, gloptipoly3:3, gurobi:8, iris:3, lcm:24, mosek:2, octomap:1, sedumi:3, yalmip:2)
 0 timed out
 33 failed
  4 <a href=\"https://github.com/RobotLocomotion/drake/issues/374\">known issue #374</a>: failing imu unit tests
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/466\">known issue #466</a>: implement Jdot*v for drakeFunction.kinematic methods
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/468\">known issue #468</a>: non-bullet builds can throw logical error in active_collision parsing in collisionDetect
  25 <a href=\"https://github.com/RobotLocomotion/drake/issues/569\">known issue #569</a>: lcppath.mexw64 and lcppath.mexw32 fails to load on win machines
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/585\">known issue #585</a>: testIndividualCentersOfPressure fails on win64
  1 unknown

solvers/test/rrt_piano_mover_rotation (unknown issue)
```
error_id = MATLAB:nonLogicalConditional
error_message = Operands to the || and && operators must be convertible to logical scalar values.
is_segfault = 0
```
```Matlab
C:/Program Files/MATLAB/R2012a/bin/win64/MATLAB.exe
 
To get started, type one of these: helpwin, helpdesk, or demo.
For product information, visit www.mathworks.com.
 
To reproduce this test use rng(415969124,'twister')
 
{Warning: Disabling autosave for simulink blocks (to avoid generating a lot of *.mdl.autosave files in your
directory.  If you aren't a regular Simulink user and don't want this disabled, comment out this section
in addpath_drake.
To disable this warning in the future, add
   warning('off','Drake:DisablingSimulinkAutosave')
to your matlab startup.m} 
{> In addpath_drake at 74
  In addpath_drake at 4} 
Operands to the || and && operators must be convertible to logical scalar values.

Error in rrt_piano_mover_rotation (line 33)
    if (collisionConstraint(r_new, q0) > min_distance) && ...

Error in fevalPackageSafe (line 21)
  feval(f);

 
<test_name>solvers/test/rrt_piano_mover_rotation</test_name> <error_id>MATLAB:nonLogicalConditional</error_id> <error_message>Operands to the || and && operators must be convertible to logical scalar values.</error_message> 


```

RobotLocomotionBuild commented:
Site: drake006
Build Name: master-pull-avalenzu-sampling-based-planning-into-RobotLocomotion-master
Build Server Directory: /tmp/drake-master-3889/drake
Pull Request: https://github.com/RobotLocomotion/drake/pull/760
CDash: http://kobol.csail.mit.edu/cd/viewTest.php?onlyfailed&buildid=10962

Most recent commits to drake included in this build:
```
5f32688 Merge branch 'sampling-based-planning' of https://github.com/avalenzu/drake
390dca5 Merge pull request #762 from RussTedrake/merge_into_drc
6a38bf6 publish drakeUtil
743c196 Add quaternionDistance.
aa52b79 Merge pull request #745 from psiorx/jc-profiler-tweaks
```

359 tests run:
 347 passed
 0 cancelled (with missing dependencies )
 1 timed out
 11 failed
  2 <a href=\"https://github.com/RobotLocomotion/drake/issues/18\">known issue #18</a>: Singularities in the euler-angle floating base joint cause infrequent errors on unit tests with random initial conditions
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/311\">known issue #311</a>: testFloatingBase test can fail to match tolerance
  4 <a href=\"https://github.com/RobotLocomotion/drake/issues/374\">known issue #374</a>: failing imu unit tests
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/466\">known issue #466</a>: implement Jdot*v for drakeFunction.kinematic methods
  3 unknown

solvers/test/rrt_piano_mover (unknown issue)
```
error_id = MATLAB:UndefinedFunction
error_message = Undefined function 'slerp' for input arguments of type 'double'.
is_segfault = 0
```
```Matlab

                            < M A T L A B (R) >
                  Copyright 1984-2012 The MathWorks, Inc.
                    R2012b (8.0.0.783) 64-bit (glnxa64)
                              August 22, 2012

 
To get started, type one of these: helpwin, helpdesk, or demo.
For product information, visit www.mathworks.com.
 
To reproduce this test use rng(416234711,'twister')
 
[Warning: Disabling autosave for simulink blocks (to avoid generating a lot of
*.mdl.autosave files in your directory.  If you aren't a regular Simulink user
and don't want this disabled, comment out this section in addpath_drake.
To disable this warning in the future, add
   warning('off','Drake:DisablingSimulinkAutosave')
to your matlab startup.m] 
[> In addpath_drake at 74
  In addpath_drake at 4] 
 Added the lcm jar to your javaclasspath (found via cmake)
 Added the lcmgl jar to your javaclasspath (found via cmake)
LCM: Disabling IPV6 support
LCM: TTL set to zero, traffic will not leave localhost.
Undefined function 'slerp' for input arguments of type 'double'.

Error in SO3MotionPlanningTree/interpolate (line 22)
      q = slerp(q1, q2, f);

Error in CompositeVertexArrayTree/interpolate (line 54)
        q(obj.idx{i},:) = obj.trees{i}.interpolate(q1(obj.idx{i}), ...

Error in MotionPlanningTree/newConfig (line 136)
        q_new = obj.interpolate(obj.getVertex(id_near), q, alpha);

Error in MotionPlanningTree/extend (line 161)
      [obj, q_new, id_near] = newConfig(obj, q);

Error in MotionPlanningTree/rrtNew (line 48)
        [obj, status] = extend(obj, x_sample);

Error in rrt_piano_mover (line 57)
      [TA, path_ids_A, info] = TA.rrtNew(q0, qf, options);

Error in fevalPackageSafe (line 21)
  feval(f);

 
<test_name>solvers/test/rrt_piano_mover</test_name> <error_id>MATLAB:UndefinedFunction</error_id> <error_message>Undefined function 'slerp' for input arguments of type 'double'.</error_message> 


```

solvers/test/rrt_piano_mover_rotation (unknown issue)
```
error_id = MATLAB:UndefinedFunction
error_message = Undefined function 'slerp' for input arguments of type 'double'.
is_segfault = 0
```
```Matlab

                            < M A T L A B (R) >
                  Copyright 1984-2012 The MathWorks, Inc.
                    R2012b (8.0.0.783) 64-bit (glnxa64)
                              August 22, 2012

 
To get started, type one of these: helpwin, helpdesk, or demo.
For product information, visit www.mathworks.com.
 
To reproduce this test use rng(416234810,'twister')
 
[Warning: Disabling autosave for simulink blocks (to avoid generating a lot of
*.mdl.autosave files in your directory.  If you aren't a regular Simulink user
and don't want this disabled, comment out this section in addpath_drake.
To disable this warning in the future, add
   warning('off','Drake:DisablingSimulinkAutosave')

**This post was going to be too long and has been truncated**

RobotLocomotionBuild commented:
Site: DRAKE003
Build Name: precompiled-pull-avalenzu-sampling-based-planning-into-RobotLocomotion-master
Build Server Directory: C:\\cygwin64\	mp\\drake-precompiled-3718\\drake
Pull Request: https://github.com/RobotLocomotion/drake/pull/760
CDash: http://kobol.csail.mit.edu/cd/viewTest.php?onlyfailed&buildid=10966

Most recent commits to drake included in this build:
```
390dca5 Merge pull request #762 from RussTedrake/merge_into_drc
6a38bf6 publish drakeUtil
aa52b79 Merge pull request #745 from psiorx/jc-profiler-tweaks
74a1c29 Merge pull request #756 from RussTedrake/playing_w_cars
ab6c084 removed extra lines (as suggest by @avalenzu)
```

348 tests run:
 251 passed
 65 cancelled (with missing dependencies  SDPSolver:9, avl:4, bullet:9, gloptipoly3:3, gurobi:8, iris:3, lcm:21, mosek:2, octomap:1, sedumi:3, yalmip:2)
 0 timed out
 32 failed
  4 <a href=\"https://github.com/RobotLocomotion/drake/issues/374\">known issue #374</a>: failing imu unit tests
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/466\">known issue #466</a>: implement Jdot*v for drakeFunction.kinematic methods
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/468\">known issue #468</a>: non-bullet builds can throw logical error in active_collision parsing in collisionDetect
  25 <a href=\"https://github.com/RobotLocomotion/drake/issues/569\">known issue #569</a>: lcppath.mexw64 and lcppath.mexw32 fails to load on win machines
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/585\">known issue #585</a>: testIndividualCentersOfPressure fails on win64
  0 unknown

RobotLocomotionBuild commented:
Site: drake006
Build Name: master-pull-avalenzu-sampling-based-planning-into-RobotLocomotion-master
Build Server Directory: /tmp/drake-master-1967/drake
Pull Request: https://github.com/RobotLocomotion/drake/pull/760
CDash: http://kobol.csail.mit.edu/cd/viewTest.php?onlyfailed&buildid=10970

Most recent commits to drake included in this build:
```
7f3ea6b Merge branch 'sampling-based-planning' of https://github.com/avalenzu/drake
da2dc3f Add slerp
390dca5 Merge pull request #762 from RussTedrake/merge_into_drc
6a38bf6 publish drakeUtil
743c196 Add quaternionDistance.
```

359 tests run:
 348 passed
 0 cancelled (with missing dependencies )
 1 timed out
 10 failed
  2 <a href=\"https://github.com/RobotLocomotion/drake/issues/18\">known issue #18</a>: Singularities in the euler-angle floating base joint cause infrequent errors on unit tests with random initial conditions
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/286\">known issue #286</a>: getPoints from stl mesh file can fail in parallel unit tests
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/311\">known issue #311</a>: testFloatingBase test can fail to match tolerance
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/313\">known issue #313</a>: planarNlink lqr can occasionally trip the zero-crossing limit
  4 <a href=\"https://github.com/RobotLocomotion/drake/issues/374\">known issue #374</a>: failing imu unit tests
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/466\">known issue #466</a>: implement Jdot*v for drakeFunction.kinematic methods
  0 unknown

solvers/test/testTaskSpaceRRT (timed out)

RobotLocomotionBuild commented:
Site: DRAKE003
Build Name: precompiled-pull-avalenzu-sampling-based-planning-into-RobotLocomotion-master
Build Server Directory: C:\\cygwin64\	mp\\drake-precompiled-28267\\drake
Pull Request: https://github.com/RobotLocomotion/drake/pull/760
CDash: http://kobol.csail.mit.edu/cd/viewTest.php?onlyfailed&buildid=10971

Most recent commits to drake included in this build:
```
52ba701 Merge branch 'sampling-based-planning' of https://github.com/avalenzu/drake
da2dc3f Add slerp
390dca5 Merge pull request #762 from RussTedrake/merge_into_drc
6a38bf6 publish drakeUtil
743c196 Add quaternionDistance.
```

352 tests run:
 251 passed
 68 cancelled (with missing dependencies  SDPSolver:9, avl:4, bullet:9, gloptipoly3:3, gurobi:8, iris:3, lcm:24, mosek:2, octomap:1, sedumi:3, yalmip:2)
 0 timed out
 33 failed
  4 <a href=\"https://github.com/RobotLocomotion/drake/issues/374\">known issue #374</a>: failing imu unit tests
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/466\">known issue #466</a>: implement Jdot*v for drakeFunction.kinematic methods
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/468\">known issue #468</a>: non-bullet builds can throw logical error in active_collision parsing in collisionDetect
  25 <a href=\"https://github.com/RobotLocomotion/drake/issues/569\">known issue #569</a>: lcppath.mexw64 and lcppath.mexw32 fails to load on win machines
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/585\">known issue #585</a>: testIndividualCentersOfPressure fails on win64
  1 unknown

solvers/test/rrt_piano_mover_rotation (unknown issue)
```
error_id = MATLAB:nonLogicalConditional
error_message = Operands to the || and && operators must be convertible to logical scalar values.
is_segfault = 0
```
```Matlab
C:/Program Files/MATLAB/R2012a/bin/win64/MATLAB.exe
 
To get started, type one of these: helpwin, helpdesk, or demo.
For product information, visit www.mathworks.com.
 
To reproduce this test use rng(418868735,'twister')
 
{Warning: Disabling autosave for simulink blocks (to avoid generating a lot of *.mdl.autosave files in your
directory.  If you aren't a regular Simulink user and don't want this disabled, comment out this section
in addpath_drake.
To disable this warning in the future, add
   warning('off','Drake:DisablingSimulinkAutosave')
to your matlab startup.m} 
{> In addpath_drake at 74
  In addpath_drake at 4} 
Operands to the || and && operators must be convertible to logical scalar values.

Error in rrt_piano_mover_rotation (line 33)
    if (collisionConstraint(r_new, q0) > min_distance) && ...

Error in fevalPackageSafe (line 21)
  feval(f);

 
<test_name>solvers/test/rrt_piano_mover_rotation</test_name> <error_id>MATLAB:nonLogicalConditional</error_id> <error_message>Operands to the || and && operators must be convertible to logical scalar values.</error_message> 


```

RobotLocomotionBuild commented:
Site: drake001.csail.mit.edu
Build Name: master-pull-avalenzu-sampling-based-planning-into-RobotLocomotion-master
Build Server Directory: /private/tmp/drake-master-13055/drake
Pull Request: https://github.com/RobotLocomotion/drake/pull/760
CDash: http://kobol.csail.mit.edu/cd/viewTest.php?onlyfailed&buildid=10975

Most recent commits to drake included in this build:
```
9730d0e Merge branch 'sampling-based-planning' of https://github.com/avalenzu/drake
390dca5 Merge pull request #762 from RussTedrake/merge_into_drc
6a38bf6 publish drakeUtil
743c196 Add quaternionDistance.
aa52b79 Merge pull request #745 from psiorx/jc-profiler-tweaks
```

356 tests run:
 342 passed
 0 cancelled (with missing dependencies )
 2 timed out
 12 failed
  3 <a href=\"https://github.com/RobotLocomotion/drake/issues/18\">known issue #18</a>: Singularities in the euler-angle floating base joint cause infrequent errors on unit tests with random initial conditions
  4 <a href=\"https://github.com/RobotLocomotion/drake/issues/374\">known issue #374</a>: failing imu unit tests
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/466\">known issue #466</a>: implement Jdot*v for drakeFunction.kinematic methods
  4 unknown

solvers/test/rrt_piano_mover (unknown issue)
```
error_id = MATLAB:UndefinedFunction
error_message = Undefined function 'slerp' for input arguments of type 'double'.
is_segfault = 0
```
```Matlab

                            < M A T L A B (R) >
                  Copyright 1984-2014 The MathWorks, Inc.
                     R2014a (8.3.0.532) 64-bit (maci64)
                             February 11, 2014

 
To get started, type one of these: helpwin, helpdesk, or demo.
For product information, visit www.mathworks.com.
 
To reproduce this test use rng(417983526,'twister')
 
[Warning: Disabling autosave for simulink blocks (to avoid generating a lot of
*.mdl.autosave files in your directory.  If you aren't a regular Simulink user
and don't want this disabled, comment out this section in addpath_drake.
To disable this warning in the future, add
   warning('off','Drake:DisablingSimulinkAutosave')
to your matlab startup.m] 
[> In addpath_drake at 74
  In addpath_drake at 4] 
 Added the lcm jar to your javaclasspath (found via cmake)
[Warning: Currently all of your LCM traffic will be broadcast to the network,
because:
  Route to multicast channel does not run through the loopback interface.
  Consider running: sudo
  /private/tmp/drake-master-13055/drake/util/setup_loopback_multicast.sh
] 
[> In checkDependency at 62
  In checkDependency at 73
  In LCMGLClient at 3
  In MotionPlanningTree>MotionPlanningTree.setLCMGL at 236
  In SE3MotionPlanningTree>SE3MotionPlanningTree.setLCMGL at 65
  In SE3MotionPlanningTree>SE3MotionPlanningTree.SE3MotionPlanningTree at 12
  In rrt_piano_mover at 23
  In fevalPackageSafe at 21] 
 Added the lcmgl jar to your javaclasspath (found via cmake)
LCM: Disabling IPV6 support
LCM: TTL set to zero, traffic will not leave localhost.
Undefined function 'slerp' for input arguments of type 'double'.

Error in SO3MotionPlanningTree/interpolate (line 22)
      q = slerp(q1, q2, f);

Error in CompositeVertexArrayTree/interpolate (line 54)
        q(obj.idx{i},:) = obj.trees{i}.interpolate(q1(obj.idx{i}), ...

Error in MotionPlanningTree/newConfig (line 136)
        q_new = obj.interpolate(obj.getVertex(id_near), q, alpha);

Error in MotionPlanningTree/extend (line 161)
      [obj, q_new, id_near] = newConfig(obj, q);

Error in MotionPlanningTree/rrtNew (line 48)
        [obj, status] = extend(obj, x_sample);

Error in rrt_piano_mover (line 57)
      [TA, path_ids_A, info] = TA.rrtNew(q0, qf, options);

Error in fevalPackageSafe (line 21)
  feval(f);

 
<test_name>solvers/test/rrt_piano_mover</test_name> <error_id>MATLAB:UndefinedFunction</error_id> <error_message>Undefined function 'slerp' for input arguments of type 'double'.</error_message> 


```

solvers/test/rrt_piano_mover_rotation (unknown issue)
```
error_id = MATLAB:UndefinedFunction
error_message = Undefined function 'slerp' for input arguments of type 'double'.
is_segfault = 0
```
```Matlab

                            < M A T L A B (R) >
                  Copyright 1984-2014 The MathWorks, Inc.
                     R2014a (8.3.0.532) 64-bit (maci64)
                             February 11, 2014

 
To get starte

**This post was going to be too long and has been truncated**

RobotLocomotionBuild commented:
Site: drake006
Build Name: master-pull-avalenzu-sampling-based-planning-into-RobotLocomotion-master
Build Server Directory: /tmp/drake-master-9878/drake
Pull Request: https://github.com/RobotLocomotion/drake/pull/760
CDash: http://kobol.csail.mit.edu/cd/viewTest.php?onlyfailed&buildid=10985

Most recent commits to drake included in this build:
```
83947ed Merge branch 'sampling-based-planning' of https://github.com/avalenzu/drake
3c58f33 Add missing \"any\"s found by Windows build server
da2dc3f Add slerp
390dca5 Merge pull request #762 from RussTedrake/merge_into_drc
6a38bf6 publish drakeUtil
```

359 tests run:
 351 passed
 0 cancelled (with missing dependencies )
 0 timed out
 8 failed
  3 <a href=\"https://github.com/RobotLocomotion/drake/issues/18\">known issue #18</a>: Singularities in the euler-angle floating base joint cause infrequent errors on unit tests with random initial conditions
  4 <a href=\"https://github.com/RobotLocomotion/drake/issues/374\">known issue #374</a>: failing imu unit tests
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/466\">known issue #466</a>: implement Jdot*v for drakeFunction.kinematic methods
  0 unknown

RobotLocomotionBuild commented:
Site: DRAKE003
Build Name: precompiled-pull-avalenzu-sampling-based-planning-into-RobotLocomotion-master
Build Server Directory: C:\\cygwin64\	mp\\drake-precompiled-7526\\drake
Pull Request: https://github.com/RobotLocomotion/drake/pull/760
CDash: http://kobol.csail.mit.edu/cd/viewTest.php?onlyfailed&buildid=10986

Most recent commits to drake included in this build:
```
724e9f0 Merge branch 'sampling-based-planning' of https://github.com/avalenzu/drake
3c58f33 Add missing \"any\"s found by Windows build server
da2dc3f Add slerp
390dca5 Merge pull request #762 from RussTedrake/merge_into_drc
6a38bf6 publish drakeUtil
```

352 tests run:
 251 passed
 69 cancelled (with missing dependencies  SDPSolver:9, avl:4, bullet:9, gloptipoly3:3, gurobi:8, iris:3, lcm:25, mosek:2, octomap:1, sedumi:3, yalmip:2)
 0 timed out
 32 failed
  4 <a href=\"https://github.com/RobotLocomotion/drake/issues/374\">known issue #374</a>: failing imu unit tests
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/466\">known issue #466</a>: implement Jdot*v for drakeFunction.kinematic methods
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/468\">known issue #468</a>: non-bullet builds can throw logical error in active_collision parsing in collisionDetect
  25 <a href=\"https://github.com/RobotLocomotion/drake/issues/569\">known issue #569</a>: lcppath.mexw64 and lcppath.mexw32 fails to load on win machines
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/585\">known issue #585</a>: testIndividualCentersOfPressure fails on win64
  0 unknown

RobotLocomotionBuild commented:
Site: DRAKE004
Build Name: precompiled-pull-avalenzu-sampling-based-planning-into-RobotLocomotion-master
Build Server Directory: C:\\cygwin\	mp\\drake-precompiled-27946\\drake
Pull Request: https://github.com/RobotLocomotion/drake/pull/760
CDash: http://kobol.csail.mit.edu/cd/viewTest.php?onlyfailed&buildid=10987

Most recent commits to drake included in this build:
```
614773c Merge branch 'sampling-based-planning' of https://github.com/avalenzu/drake
3c58f33 Add missing \"any\"s found by Windows build server
da2dc3f Add slerp
390dca5 Merge pull request #762 from RussTedrake/merge_into_drc
6a38bf6 publish drakeUtil
```

352 tests run:
 250 passed
 69 cancelled (with missing dependencies  SDPSolver:9, avl:4, bullet:9, gloptipoly3:3, gurobi:8, iris:3, lcm:25, mosek:2, octomap:1, sedumi:3, yalmip:2)
 0 timed out
 33 failed
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/366\">known issue #366</a>: linear_gaussian_dt fails with small probability
  4 <a href=\"https://github.com/RobotLocomotion/drake/issues/374\">known issue #374</a>: failing imu unit tests
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/466\">known issue #466</a>: implement Jdot*v for drakeFunction.kinematic methods
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/468\">known issue #468</a>: non-bullet builds can throw logical error in active_collision parsing in collisionDetect
  25 <a href=\"https://github.com/RobotLocomotion/drake/issues/569\">known issue #569</a>: lcppath.mexw64 and lcppath.mexw32 fails to load on win machines
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/585\">known issue #585</a>: testIndividualCentersOfPressure fails on win64
  0 unknown

RobotLocomotionBuild commented:
Site: drake001.csail.mit.edu
Build Name: master-pull-avalenzu-sampling-based-planning-into-RobotLocomotion-master
Build Server Directory: /private/tmp/drake-master-5553/drake
Pull Request: https://github.com/RobotLocomotion/drake/pull/760
CDash: http://kobol.csail.mit.edu/cd/viewTest.php?onlyfailed&buildid=11001

Most recent commits to drake included in this build:
```
df2aa06 Merge branch 'sampling-based-planning' of https://github.com/avalenzu/drake
3c58f33 Add missing \"any\"s found by Windows build server
da2dc3f Add slerp
390dca5 Merge pull request #762 from RussTedrake/merge_into_drc
6a38bf6 publish drakeUtil
```

356 tests run:
 346 passed
 0 cancelled (with missing dependencies )
 3 timed out
 7 failed
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/18\">known issue #18</a>: Singularities in the euler-angle floating base joint cause infrequent errors on unit tests with random initial conditions
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/311\">known issue #311</a>: testFloatingBase test can fail to match tolerance
  4 <a href=\"https://github.com/RobotLocomotion/drake/issues/374\">known issue #374</a>: failing imu unit tests
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/466\">known issue #466</a>: implement Jdot*v for drakeFunction.kinematic methods
  0 unknown

examples/Atlas/test/testRunningPlanner (timed out)

examples/DubinsCar/runFunnel (timed out)

examples/UnderwaterAcrobot/runSwingUp (timed out)

avalenzu commented:
OK, we've got a clean bill of health from the build servers now. This is
just pending code review.

On Sat, Feb 14, 2015, 8:46 PM RobotLocomotionBuild <notifications@github.com>
wrote:

> Site: drake001.csail.mit.edu
> Build Name:
> master-pull-avalenzu-sampling-based-planning-into-RobotLocomotion-master
> Build Server Directory: /private/tmp/drake-master-5553/drake
> Pull Request: #760 <https://github.com/RobotLocomotion/drake/pull/760>
> CDash: http://kobol.csail.mit.edu/cd/viewTest.php?onlyfailed&buildid=11001
>
> Most recent commits to drake included in this build:
>
> df2aa06 Merge branch 'sampling-based-planning' of https://github.com/avalenzu/drake
> 3c58f33 <https://github.com/avalenzu/drake3c58f33> Add missing \"any\"s found by Windows build server
> da2dc3f Add slerp
> 390dca5 Merge pull request #762 from RussTedrake/merge_into_drc
> 6a38bf6 publish drakeUtil
>
> 356 tests run:
> 346 passed
>
>
> 0 cancelled (with missing dependencies )
> 3 timed out
> 7 failed
> 1 known issue #18 <https://github.com/RobotLocomotion/drake/issues/18>:
> Singularities in the euler-angle floating base joint cause infrequent
> errors on unit tests with random initial conditions
>
> 1 known issue #311 <https://github.com/RobotLocomotion/drake/issues/311>:
> testFloatingBase test can fail to match tolerance
>
> 4 known issue #374 <https://github.com/RobotLocomotion/drake/issues/374>:
> failing imu unit tests
> 1 known issue #466 <https://github.com/RobotLocomotion/drake/issues/466>:
> implement Jdot*v for drakeFunction.kinematic methods
>
> 0 unknown
>
> examples/Atlas/test/testRunningPlanner (timed out)
>
> examples/DubinsCar/runFunnel (timed out)
>
> examples/UnderwaterAcrobot/runSwingUp (timed out)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RobotLocomotion/drake/pull/760#issuecomment-74400270>.
>

avalenzu commented:
@RussTedrake and/or @patmarion could I get a code review for this one? Thanks!

RussTedrake commented on solvers/+drakeFunction/+kinematic/SmoothDistancePenalty.m:
```diff
@@ -29,12 +29,8 @@
-      dcost_dscaled_dist = sum(dpairwise_cost_dscaled_dist,1);
-      df = dcost_dscaled_dist*dscaled_dist_ddist*ddist_dq;
+      obj.rbm.doKinematics(q);
+      [f,df] = smoothDistancePenaltymex(obj.rbm.mex_model_ptr,obj.min_distance,false,obj.active_collision_options);
```
this won't work for builds w/o the mex_model_ptr.

RussTedrake commented on solvers/CartesianMotionPlanningTree.m:
```diff
@@ -0,0 +1,34 @@
+classdef CartesianMotionPlanningTree < VertexArrayMotionPlanningTree
```
at least a little documentation please

RussTedrake commented on solvers/CartesianMotionPlanningTree.m:
```diff
@@ -0,0 +1,34 @@
+      obj = obj@VertexArrayMotionPlanningTree(num_vars);
+    end
+
+    function q = interpolate(obj, q1, q2, f)
```
f is a strange name for it.  doc please?

RussTedrake commented on solvers/VertexArrayMotionPlanningTree.m:
```diff
@@ -0,0 +1,59 @@
+  % "checkConstraints" functionality of MotionPlanningProblem and stores the
+  % vertices of the tree in a num_vars-by-n array.
+  properties
+    lb
```
doc?  presumably this is just for your sampling strategy?

RussTedrake commented on solvers/MotionPlanningTree.m:
```diff
@@ -0,0 +1,273 @@
+  end
+
+  methods (Abstract)
+    q = randomConfig(obj);
```
it's not always a configuration... right?  randomSample might be better?

RussTedrake commented on solvers/MotionPlanningTree.m:
```diff
@@ -0,0 +1,273 @@
+      obj.n = 0;
+    end
+
+    function [obj, path_ids, info] = rrtNew(obj, x_start, x_goal, options)
```
a little doc?  (why New?)

avalenzu replied:
Sorry, the new was just to prevent name clashes before I removed the `rrt` method from `MotionPlanningProblem`. I'll rename and add doc.

RussTedrake commented on solvers/CompositeVertexArrayTree.m:
```diff
@@ -0,0 +1,89 @@
+classdef CompositeVertexArrayTree < VertexArrayMotionPlanningTree
```
doc here, too.   why do I want multiple weighted trees, exactly?

RussTedrake commented on solvers/ConfigurationSpaceMotionPlanningTree.m:
```diff
@@ -0,0 +1,131 @@
+classdef ConfigurationSpaceMotionPlanningTree < CartesianMotionPlanningTree
```
a configuration space is not a cartesian space...?

avalenzu replied:
Agreed. I'll rename this class to `JointSpaceMotionPlanningTree`, since we usually treat the joint space as a Cartesian space.

RussTedrake commented on solvers/ConfigurationSpaceMotionPlanningTree.m:
```diff
@@ -0,0 +1,131 @@
+    active_collision_options = struct();
+    v
+    xyz_vis
+    ik_constraints = {};
```
there really kinematic constraints (not ik)

RussTedrake commented on solvers/ConfigurationSpaceMotionPlanningTree.m:
```diff
@@ -0,0 +1,131 @@
+      obj.xyz_vis = NaN(3, obj.N);
+    end
+
+    function obj = addIKConstraint(obj, varargin)
```
again i'd say they're kinematic constraints. (not IK)

RussTedrake commented on solvers/ConfigurationSpaceMotionPlanningTree.m:
```diff
@@ -0,0 +1,131 @@
+      valid = (info < 10);
+    end
+
+    function valid = checkIKConstraints(obj, q)
```
Kin?  :)

RussTedrake commented on solvers/MotionPlanningTree.m:
```diff
@@ -0,0 +1,273 @@
+    end
+
+
+    function [obj, q_new, id_near] = newConfig(obj, q)
```
again, not necessarily config...?

avalenzu replied:
`newPoint`?

RussTedrake replied:
i'm ok with that.
On Feb 16, 2015, at 2:46 PM, Andres Valenzuela <notifications@github.com> wrote:

> In solvers/MotionPlanningTree.m:
> 
> > +%       [TA, status, id_new] = connect(TA, x_sample);
> > +      path_ids_A = [];
> > +      path_ids_B = [];
> > +      if status == TA.TRAPPED && ~isempty(id_new), keyboard; end
> > +      if status ~= TA.TRAPPED && rand < goal_bias
> > +        [TB, status, id_last] = connect(TB, TA.getVertex(id_new));
> > +
> > +        if status == TB.REACHED
> > +          path_ids_A = TA.getPathToVertex(id_new);
> > +          path_ids_B = TB.getPathToVertex(id_last);
> > +        end
> > +      end
> > +    end
> > +
> > +
> > +    function [obj, q_new, id_near] = newConfig(obj, q)
> newPoint?
> 
> —
> Reply to this email directly or view it on GitHub.
>

RussTedrake commented on solvers/MotionPlanningTree.m:
```diff
@@ -0,0 +1,273 @@
+      end
+    end
+
+    function [q, valid] = attemptToMakeValidConfiguration(obj, q, valid)
```
config

RussTedrake commented on solvers/SE3MotionPlanningTree.m:
```diff
@@ -0,0 +1,81 @@
+    function obj = SE3MotionPlanningTree()
+      T_R3 = R3MotionPlanningTree();
+      T_SO3 = SO3MotionPlanningTree();
+      obj = obj@CompositeVertexArrayTree({T_R3, T_SO3},{'R3', 'SO3'});
```
why this, instead of a single tree with all coordinates?

avalenzu replied:
I felt like this better reflected the structure of the space. Also, it lets me reuse the distance metrics already defined in those classes.

RussTedrake commented on solvers/SE3MotionPlanningTree.m:
```diff
@@ -0,0 +1,81 @@
+      obj.rbm = obj.rbm.addGeometryToBody(1, geom);
+    end
+
+    function obj = setOrientationWeight(obj, orientation_weight)
```
the weights feel like a redundant encoding on top of the distance metric...

avalenzu replied:
I'm not sure I follow. The weights are a part of the distance metric.

```matlab
d([x1; y1], [x2; y2]) = weights(1)*d_x(x1, x2) + weights(2)*d_y(y1, y2)
```

avalenzu replied:
I'm drawing this treatment from [Kuffner's paper on sampling and distance metrics in SE(3)](http://www.cs.cmu.edu/afs/cs/academic/class/15494-s14/readings/kuffner_icra2000.pdf).

RussTedrake commented on systems/plants/constraint/QuasiStaticConstraint.m:
```diff
@@ -165,18 +165,9 @@
-      flag = resnorm<1e-6;
+
+      K = convhull(shrink_vertices(1,:), shrink_vertices(2,:));
+      flag = inpolygon(com(1), com(2), shrink_vertices(1,K), shrink_vertices(2,K));
```
:)

RussTedrake commented:
otherwise, looks good.  very exciting to bring this in!!

avalenzu commented:
Thanks for all the comments! I'll address those and update the pull request.

On Mon Feb 16 2015 at 11:08:08 AM Russ Tedrake <notifications@github.com>
wrote:

> otherwise, looks good. very exciting to bring this in!!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RobotLocomotion/drake/pull/760#issuecomment-74531414>.
>

avalenzu commented:
I've addressed @RussTedrake's comments and added some additional documentation. There was a fair amount of method renaming. The tests pass on my machine, now let's see what the build servers think.

RussTedrake commented:
the changes all look good to me.  will accept if the build servers say "OK".

avalenzu commented:
Excellent. Thanks!

On Mon, Feb 16, 2015, 6:26 PM Russ Tedrake <notifications@github.com> wrote:

> the changes all look good to me. will accept if the build servers say "OK".
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RobotLocomotion/drake/pull/760#issuecomment-74589179>.
>

RobotLocomotionBuild commented:
Site: drake006
Build Name: master-pull-avalenzu-sampling-based-planning-into-RobotLocomotion-master
Build Server Directory: /tmp/drake-master-250/drake
Pull Request: https://github.com/RobotLocomotion/drake/pull/760
CDash: http://kobol.csail.mit.edu/cd/viewTest.php?onlyfailed&buildid=11060

Most recent commits to drake included in this build:
```
da9fb17 Merge branch 'sampling-based-planning' of https://github.com/avalenzu/drake
ad4ffc3 Fix place where MotionPlanningTree was tied to one representation
cddd5bb Make changes to address @RussTedrake's comments
4b539ec Fix SmoothDistancePenalty for non-mex builds
748f6c9 Merge pull request #765 from RussTedrake/inertia_in_urdf_cpp
```

360 tests run:
 347 passed
 0 cancelled (with missing dependencies )
 2 timed out
 11 failed
  2 <a href=\"https://github.com/RobotLocomotion/drake/issues/18\">known issue #18</a>: Singularities in the euler-angle floating base joint cause infrequent errors on unit tests with random initial conditions
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/310\">known issue #310</a>: testLCMPlant fails on parallel build because it can't keep up with realtime
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/311\">known issue #311</a>: testFloatingBase test can fail to match tolerance
  4 <a href=\"https://github.com/RobotLocomotion/drake/issues/374\">known issue #374</a>: failing imu unit tests
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/466\">known issue #466</a>: implement Jdot*v for drakeFunction.kinematic methods
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/766\">known issue #766</a>: new forwardKin gives nans in jacobians
  1 unknown

systems/plants/test/testForwardKin (unknown issue)
```
error_id = Drake:ValueCheck
error_message = Values don't match.  Expected 
[1.39706212691382 0.982932283798885 -0.179440634922389 0.400673174035788 0.000307679966546903;0.0659853492088657 0.415979545234059 1.43816405956632 0.984881801517688 -1.32229448619544;-0.925506492945206 0.897273646432425 1.06739875173278 -1.18220109265525 -0.880989570227474;2.18256881272747 2.18256881272747 2.18256881272747 2.18256881272747 2.18256881272747;-0.680624221941156 -0.680624221941156 -0.680624221941156 -0.680624221941156 -0.680624221941156;3.14159265358979 3.14159265358979 3.14159265358979 3.14159265358979 3.14159265358979]
 but got 
[1.39706212691382 0.982932283798885 -0.179440634922389 0.400673174035788 0.000307679966546792;0.0659853492088658 0.415979545234059 1.43816405956632 0.984881801517686 -1.32229448619543;-0.925506492945205 0.897273646432425 1.06739875173278 -1.18220109265525 -0.880989570227473;2.18256881272747 2.18256881272747 2.18256881272747 2.18256881272747 2.18256881272747;-0.680624221941156 -0.680624221941156 -0.680624221941156 -0.680624221941156 -0.680624221941156;-3.14159265358979 -3.14159265358979 -3.14159265358979 -3.14159265358979 -3.14159265358979]
is_segfault = 0
```
```Matlab

                            < M A T L A B (R) >
                  Copyright 1984-2012 The MathWorks, Inc.
                    R2012b (8.0.0.783) 64-bit (glnxa64)
                              August 22, 2012

 
To get started, type one of these: helpwin, helpdesk, or demo.
For product information, visit www.mathworks.com.
 
To reproduce this test use rng(442483817,'twister')
 
[Warning: Disabling autosave for simulink blocks (to avoid generating a lot of
*.mdl.autosave files in your directory.  If you aren't a regular Simulink user
and don't want this disabled, comment out this section in addpath_drake.
To disable this warning in the future, add
   warning('off','Drake:DisablingSimulinkAutosave')
to your matlab startup.m] 
[> In addpath_drake at 74
  In addpath_drake at 4] 
Error using valuecheck (line 78)
Values don't match.  Expected 
[1.39706212691382 0.982932283798885 -0.179440634922389 0.400673174035788 0.000307679966546903;0.0659853492088657 0.415979545234059 1.43816405956632 0.984881801517688 -1.32229448619544;-0.925506492945206 0.897273646432425 1.06739875173278 -1.18220109265525 -0.880989570227474;2.18256881272747 2.18256881272747 2.18256881272747 2.18256881272747 2.18256881272747;-0.680624221941156 -0.680624221941156 -0.680624221941156 -0.680624221941156 -0.680624221941156;3.14159265358979 3.14159265358979 3.14159265358979 3.14159265358979 3.14159265358979]
 but got 
[1.39706212691382 0.982932283798885 -0.17944063

**This post was going to be too long and has been truncated**

RussTedrake commented:
this one is not your fault.  a few of the kinematics tests have started failing with off by 2*pi errors.  @tkoolen - i think it's a difference with the kinsols, right?  can you make the unit test more robust?

On Feb 16, 2015, at 6:54 PM, RobotLocomotionBuild <notifications@github.com> wrote:

> Site: drake006
> Build Name: master-pull-avalenzu-sampling-based-planning-into-RobotLocomotion-master
> Build Server Directory: /tmp/drake-master-250/drake
> Pull Request: #760
> CDash: http://kobol.csail.mit.edu/cd/viewTest.php?onlyfailed&buildid=11060
> 
> Most recent commits to drake included in this build:
> 
> da9fb17 Merge branch 'sampling-based-planning' of https://github.com/avalenzu/drake
> ad4ffc3 Fix place where MotionPlanningTree was tied to one representation
> cddd5bb Make changes to address @RussTedrake's comments
> 4b539ec Fix SmoothDistancePenalty for non-mex builds
> 748f6c9 Merge pull request #765 from RussTedrake/inertia_in_urdf_cpp
> 360 tests run:
> 347 passed
> 0 cancelled (with missing dependencies )
> 2 timed out
> 11 failed
> 2 known issue #18: Singularities in the euler-angle floating base joint cause infrequent errors on unit tests with random initial conditions
> 1 known issue #310: testLCMPlant fails on parallel build because it can't keep up with realtime
> 1 known issue #311: testFloatingBase test can fail to match tolerance
> 4 known issue #374: failing imu unit tests
> 1 known issue #466: implement Jdot*v for drakeFunction.kinematic methods
> 1 known issue #766: new forwardKin gives nans in jacobians
> 1 unknown
> 
> systems/plants/test/testForwardKin (unknown issue)
> 
> error_id = Drake:ValueCheck
> error_message = Values don't match.  Expected 
> [1.39706212691382 0.982932283798885 -0.179440634922389 0.400673174035788 0.000307679966546903;0.0659853492088657 0.415979545234059 1.43816405956632 0.984881801517688 -1.32229448619544;-0.925506492945206 0.897273646432425 1.06739875173278 -1.18220109265525 -0.880989570227474;2.18256881272747 2.18256881272747 2.18256881272747 2.18256881272747 2.18256881272747;-0.680624221941156 -0.680624221941156 -0.680624221941156 -0.680624221941156 -0.680624221941156;3.14159265358979 3.14159265358979 3.14159265358979 3.14159265358979 3.14159265358979]
>  but got 
> [1.39706212691382 0.982932283798885 -0.179440634922389 0.400673174035788 0.000307679966546792;0.0659853492088658 0.415979545234059 1.43816405956632 0.984881801517686 -1.32229448619543;-0.925506492945205 0.897273646432425 1.06739875173278 -1.18220109265525 -0.880989570227473;2.18256881272747 2.18256881272747 2.18256881272747 2.18256881272747 2.18256881272747;-0.680624221941156 -0.680624221941156 -0.680624221941156 -0.680624221941156 -0.680624221941156;-3.14159265358979 -3.14159265358979 -3.14159265358979 -3.14159265358979 -3.14159265358979]
> is_segfault = 0
>                             < M A T L A B (R) >
>                   Copyright 1984-2012 The MathWorks, Inc.
>                     R2012b (8.0.0.783) 64-bit (glnxa64)
>                               August 22, 2012
> 
> 
> To get started, type one of these: helpwin, helpdesk, or demo.
> For product information, visit www.mathworks.com.
> 
> To reproduce this test use rng(442483817,'twister')
> 
> [Warning: Disabling autosave for simulink blocks (to avoid generating a lot of
> *.mdl.autosave files in your directory.  If you aren't a regular Simulink user
> and don't want this disabled, comment out this section in addpath_drake.
> To disable this warning in the future, add
>    warning('off','Drake:DisablingSimulinkAutosave')
> to your matlab startup.m] 
> [> In addpath_drake at 74
>   In addpath_drake at 4] 
> Error using valuecheck (line 78)
> Values don't match.  Expected 
> [1.39706212691382 0.982932283798885 -0.179440634922389 0.400673174035788 0.000307679966546903;0.0659853492088657 0.415979545234059 1.43816405956632 0.984881801517688 -1.32229448619544;-0.925506492945206 0.897273646432425 1.06739875173278 -1.18220109265525 -0.880989570227474;2.18256881272747 2.18256881272747 2.18256881272747 2.18256881272747 2.18256881272747;-0.680624221941156 -0.680624221941156 -0.680624221941156 -0.680624221941156 -0.680624221941156;3.14159265358979 3.14159265358979 3.14159265358979 3.14159265358979 3.14159265358979]
>  but got 
> [1.39706212691382 0.982932283798885 -0.17944063
> 
> **This post was going to be too long and has been truncated**
> —
> Reply to this email directly or view it on GitHub.
>

RobotLocomotionBuild commented:
Site: drake001.csail.mit.edu
Build Name: master-pull-avalenzu-sampling-based-planning-into-RobotLocomotion-master
Build Server Directory: /private/tmp/drake-master-14411/drake
Pull Request: https://github.com/RobotLocomotion/drake/pull/760
CDash: http://kobol.csail.mit.edu/cd/viewTest.php?onlyfailed&buildid=11088

Most recent commits to drake included in this build:
```
4632163 Merge branch 'sampling-based-planning' of https://github.com/avalenzu/drake
ad4ffc3 Fix place where MotionPlanningTree was tied to one representation
cddd5bb Make changes to address @RussTedrake's comments
4b539ec Fix SmoothDistancePenalty for non-mex builds
748f6c9 Merge pull request #765 from RussTedrake/inertia_in_urdf_cpp
```

357 tests run:
 345 passed
 0 cancelled (with missing dependencies )
 2 timed out
 10 failed
  2 <a href=\"https://github.com/RobotLocomotion/drake/issues/18\">known issue #18</a>: Singularities in the euler-angle floating base joint cause infrequent errors on unit tests with random initial conditions
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/313\">known issue #313</a>: planarNlink lqr can occasionally trip the zero-crossing limit
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/365\">known issue #365</a>: sampledFiniteTimeVerification does not handle infeasible programs gracefully
  4 <a href=\"https://github.com/RobotLocomotion/drake/issues/374\">known issue #374</a>: failing imu unit tests
  1 <a href=\"https://github.com/RobotLocomotion/drake/issues/466\">known issue #466</a>: implement Jdot*v for drakeFunction.kinematic methods
  1 unknown

systems/plants/test/testURDFmex (unknown issue)
```
error_id = Drake:ValueCheck
error_message = Values don't match.
    Index       Value       Desired Value
   -------     --------    ---------------
     (8,1)    -0.041022        0.000000
     (9,1)    -0.026822        0.000000
    (10,1)     0.041022        0.000000
    (11,1)     0.040477        0.000000
    (20,2)     0.071647        0.000000
    (21,2)     0.046845        0.000000
    (22,2)    -0.071647        0.000000
    (23,2)    -0.014344        0.000000
   (14,12)     0.029966        0.000000
   (15,12)    -0.029966        0.000000
   (16,12)    -0.029966        0.000000
   (20,12)     0.015236        0.000000
   (19,14)    -0.041022        0.000000
   (20,14)    -0.026822        0.000000
   (21,14)     0.041022        0.000000
   (22,14)     0.040477        0.000000
    (4,15)     0.071647        0.000000
    (5,15)     0.046845        0.000000
    (6,15)    -0.071647        0.000000
    (7,15)    -0.014344        0.000000
   (18,21)    -0.051700        0.000000
   (19,21)    -0.029480        0.000000
   (20,21)     0.012700        0.000000
   (21,21)     0.012700        0.000000
   (25,23)     0.029966        0.000000
   (26,23)    -0.029966        0.000000
   (27,23)    -0.029966        0.000000
    (4,24)     0.015236        0.000000
    (3,26)    -0.041022        0.000000
    (4,26)    -0.026822        0.000000
    (5,26)     0.041022        0.000000
    (6,26)     0.040477        0.000000
    (2,28)    -0.051700        0.000000
    (3,28)    -0.029480        0.000000
    (4,28)     0.012700        0.000000
    (5,28)     0.012700        0.000000
    (9,35)     0.029966        0.000000
   (10,35)    -0.029966        0.000000
   (11,35)    -0.029966        0.000000
   (15,35)     0.015236        0.000000

is_segfault = 0
```
```Matlab

                            < M A T L A B (R) >
                  Copyright 1984-2014 The MathWorks, Inc.
                     R2014a (8.3.0.532) 64-bit (maci64)
                             February 11, 2014

 
To get started, type one of these: helpwin, helpdesk, or demo.
For product information, visit www.mathworks.com.
 
To reproduce this test use rng(446977626,'twister')
 
[Warning: Disabling autosave for simulink blocks (to avoid generating a lot of
*.mdl.autosave files in your directory.  If you aren't a regular Simulink user
and don't want this disabled, comment out this section in addpath_drake.
To disable this warning in the future, add
   warning('off','Drake:DisablingSimulinkAutosave')
to your matlab startup.m] 
[> In addpath_drake at 74
  In addpath_drake at 4] 
testing /private/tmp/drake-master-14411/drake/systems/plants/test/FallingBrick.urdf
testing /private/tmp/drake-master-14411/drake/examples/FurutaP

**This post was going to be too long and has been truncated**
